### PR TITLE
fix #613 for queries with join from is a source

### LINKED ISSUE #613: Parameter 2 of TranslationStrategyInterface::alterQueryForTranslation should be of type SourceInterface
otherwise this fails when you have a query with a join.

### PR DESCRIPTION
**Pull request by uwej711**
see #613

### CODE REVIEW

File: lib/Doctrine/ODM/PHPCR/Translation/TranslationStrategy/TranslationStrategyInterface.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ public function getTranslatedPropertyPath($alias, $propertyName, $locale);
      */
     public function alterQueryForTranslation(
         QueryObjectModelFactoryInterface $qomf,
-        SelectorInterface &$selector,
+        SourceInterface &$selector,
```
Comment (dbu) [inline lib/Doctrine/ODM/PHPCR/Translation/TranslationStrategy/TranslationStrategyInterface.php]: this is a BC break if somebody has his own translation strategy. can you add a note on this to CHANGELOG.md please? i think we still need to break it - its not very likely people have their own strategy i hope, and we really should fix this.

Comment (uwej711) [inline lib/Doctrine/ODM/PHPCR/Translation/TranslationStrategy/TranslationStrategyInterface.php]: Done, not sure about the version ...

**Comment by dbu** [comment]
changlog looks good to me.

@lsmith77 @dantleech can we live with this BC violation? its not elegant, but i think we really need to fix this, and afaik nobody is maintaining a public custom translation strategy. in a custom project, the change is simple to adapt to, it would only be a pain to support phpcr-odm 1.2 and 1.3 at the same time.

**Comment by dantleech** [comment]
Yeah, technically we shouldn't do this. But I guess the chances are quite low that people use a custom translation strategy, its not configurable from the bundle from what I can see.

**Comment by lsmith77** [comment]
+1 on the change as the fix for users is quite simple

**Comment by dbu** [comment]
great, thanks uwe!

**Pull request merged**